deps.ffmpeg/99-ffmpeg.ps1

```diff
@@ -0,0 +1,161 @@
+    [string] $Hash = "a6dc92968a325d331bb6dcf9b3b2248026cd1d6c",
+    [array] $Patches = @(
+        @{
+            PatchFile = "${PSScriptRoot}/patches/FFmpeg/0001-flvdec-handle-unknown-Windows.patch"
+            HashSum = "72f41d25f709b1566aecaff0204e94af79d91b7845165deb5bf234440962b2fc"
+        }
+        @{
+            PatchFile = "${PSScriptRoot}/patches/FFmpeg/0002-libaomenc-presets-Windows.patch"
+            HashSum = "cec898b957fc289512094fc2c4e6a61d6872f716e4a643fb970c599a453a33f4"
+        }
+    )
+)
+
+function Setup {
+    Setup-Dependency -Uri $Uri -Hash $Hash -DestinationPath $Path
+
+    if ( ! ( $SkipAll -or $SkipDeps ) ) {
+        Invoke-External pacman.exe -S --noconfirm --needed --noprogressbar yasm
+        Invoke-External pacman.exe -S --noconfirm --needed --noprogressbar make
+        Invoke-External pacman.exe -S --noconfirm --needed --noprogressbar perl
+        Invoke-External pacman.exe -S --noconfirm --needed --noprogressbar pkgconf
+    }
+}
+
+function Clean {
+    Set-Location $Path
+
+    if ( Test-Path "build_${Target}" ) {
+        Log-Information "Clean build directory (${Target})"
+        Remove-Item -Path "build_${Target}" -Recurse -Force
+    }
+}
+
+function Patch {
+    Log-Information "Patch (${Target})"
+    Set-Location $Path
+
+    $Patches | ForEach-Object {
+        $Params = $_
+        Safe-Patch @Params
+    }
+}
+
+function Configure {
+    Log-Information "Configure (${Target})"
+    Set-Location $Path
+
+    $TargetCPUs = @{
+        x64 = 'x86_64'
+        x86 = 'x86'
+    }
+
+    New-Item -ItemType Directory -Force "build_${Target}" > $null
+
+    $ConfigureCommand = @(
+        'bash'
+        '../configure'
+        ('--prefix="' + $($script:ConfigData.OutputPath -replace '([A-Fa-f]):','/$1' -replace '\\','/') + '"')
+        ('--arch=' + $($TargetCPUs[$Target]))
+        '--toolchain=msvc'
+        ('--extra-cflags=' + "'-D_WINDLL -MD -DWINAPI_FAMILY=WINAPI_FAMILY_APP -D_WIN32_WINNT=0x0A00'")
+        ('--extra-cxxflags=' + "'-MD -DWINAPI_FAMILY=WINAPI_FAMILY_APP -D_WIN32_WINNT=0x0A00'")
+        ('--extra-ldflags=' + "'-APPCONTAINER:NO -MACHINE:${Target}'")
+        '--pkg-config=pkg-config'
+        $(if ( $Target -eq 'x64' ) { '--target-os=win64' } else { '--target-os=win32' })
+        $(if ( $Target -eq 'x64' ) { '--enable-libaom' })
+        $(if ( $Target -eq 'x64' ) { '--enable-libsvtav1' })
+        '--enable-libtheora'
+        '--enable-libmp3lame'
+        '--enable-w32threads'
+        '--enable-version3'
+        '--enable-gpl'
+        '--enable-libx264'
+        '--enable-libopus'
+        '--enable-libvorbis'
+        '--enable-libvpx'
+        '--enable-librist'
+        '--enable-libsrt'
+        '--enable-shared'
+        '--enable-zlib'
+        '--disable-static'
+        '--disable-libjack'
+        '--disable-indev=jack'
+        '--disable-sdl2'
+        '--disable-doc'
+        '--disable-postproc'
+        '--disable-mediafoundation'
```

Comment:
Is this still necessary? We might want to use it in the future for WoA and it's also usable for hardware decoding.

Reply:
We definitely need it for ARM64, but for now I preferred to keep the scope to "migrate existing builds to Windows" without bigger changes to the overall configuration (except where it was necessary to get it to compile at all).

Reply:
We could add a if not Windows arm64 then add this config.

Reply:
That is out of scope of this PR - #190 exists for that.

Reply:
As already mentioned by @PatTheMav , I'd prefer if the PR scope is kept minimal.